### DISCUSSION TITLE
Avoid issuing quests for anything inside a military base

### DISCUSSION THREAD

**Discussion by Vectorial1024** [streetcomplete/StreetComplete#5401] [Feedback]
See title.

To avoid legal troubles, sometimes military bases in OSM are tagged with a lot of deliberate "missing info"; e.g. buildings inside the military base might be tagged with simply `building=yes`, no further information about the type of building is provided (even though it is contextually clear the buildings carry an implicit `building=military` tag).

Then, in the rare case that a user is somehow near/inside a military base, this app should really avoid generating quests that attempts to clarify the details of features inside military bases.

**Comment by matkoniecz** [comment]
why specifying building type, at mapped military base, with detail available for StreetComplete mappers, would cause any type of problems?

**Reply by Vectorial1024** [reply]
I mean... the main thing about this app is "oh I am nearby and I can clarify nearby features by having a look", which kinda imply "walking closer to military bases". This sometimes does not work when the bases are closed off from the public and trpassing can carry legal penalties.

Let's not go into the alternative rabbit hole of "why I can't choose military as the category of buildings"... if we are going to cater for the entire world, then some jurisdictions will not like detailed mapping of military bases.

What _may_ work is to ask whether the given military base as an area is accessible by the public (may or may not be conditional access), but not individual buildings.

**Comment by HolgerJeromin** [comment]
We skip most quests for private things like streets with `(access !~ private|no or (foot and foot !~ private|no))`.
I think @Vectorial1024 wants to have the same to buildings if the surrounding area is access restricted.

**Reply by jonpsp** [reply]
Whilst I agree with skipping quests for military areas, I think there should be an exception (i.e. don't skip quests) for highways in a military area already explicitly tagged with an access tag of *=permissive/yes/designated. At least in my part of the UK, the public are allowed access to the countryside in some military training areas, usually only on foot but sometimes other access too.

**Reply by Vectorial1024** [reply]
I can show some real-life examples from Hong Kong, where military bases are strictly military-only, and the public only knows about the buildings location (via satellite/official maps) but not anything more.

I mean, unless someone tags the place with e.g. `permissive`/`yes`/sth else through e.g. the official iD Editor, we really should not ask for clarification of military stuff, especially when we are talking about "oh I am nearby" clarifications.

**Comment by mnalis** [comment]
Few points:
- I'm no expert (so could be wrong), but I _don't think_ it is currently in StreetComplete technically possible to (with any reliability) detect whether the building is inside military area, as it would be slow and require that whole are where the military polygon is is fully loaded first (e.g. we've had situations where we wanted to modify quests depending if they are inside urban areas or not e.g. `landuse=residential` or `natural=forest` and those turned out not to be feasible either). That seems the **main issue** to me - if it is technically not really feasible, there is little point arguing why it might be useful. (which I _do_ agree that it would be nice to add, *if* it was trivial to add _and_ did not bring any slowdowns).
- the app (when first installed) explicitly warns _"Finally, remember to stay safe: Be aware of your surroundings and **don't enter private property**"_. I would think that problematic _"military areas"_ are implicit in that _"private property"_ (i.e. private = anything that is not freely available for use by public) and that (if anything) the **users would avoid entering protected military bases without invitation _even more_** than e.g. trying locks on private houses and entering them to map their rooms for [indoor mapping](https://wiki.openstreetmap.org/wiki/Indoor_Mapping)!
- as an aside; regarding _"To avoid legal troubles, sometimes military bases in OSM are tagged with a lot of deliberate "missing info"; e.g. buildings inside the military base might be tagged with simply building=yes"_ one would expect that if there is legal trouble, one would only mark approximate military area to avoid, and **refrain from drawing any specific buildings**, roads, airstrips etc. inside military bases **at all**. But that should really be discussed within that specific community and decision reached (i.e. should any military infrastructure details be added or deleted from the OSM map?).
- as an **workaround**; especially if one thinks there is real risk SC users are likely to be shot or prosecuted by military (e.g. because SC gamification might be too addictive lure?), one could [preventively find](http://overpass-turbo.eu/s/1ETM) `landuse=military` areas, and change all `building=yes` to `building=military` as well as add `access=private` etc. to all `highway=*` inside. That would significantly reduce the user risk not only in SC, but would help other on-the-site editors too (EveryDoor, Vespucci, MapComplete, ...) as well as regular routers (OsmAnd, Organic Maps, ...). Especially as military bases are relatively few in number, and not being created every other day, that should not be too much work. But one should get the consensus in their local community first, especially if they intend to do [automated edits](https://wiki.openstreetmap.org/wiki/Automated_Edits_code_of_conduct).
